frontend/src/components/SideBar/AddGroupTags.tsx

```diff
@@ -25,15 +30,19 @@ export default function AddGroupTags({ show, setShow }: AddGroupTagsProps): Reac
           onMouseEnter={() => setShow(true)}
           onMouseLeave={() => setShow(false)}
         >
-          {classes.map((t): React.ReactElement | undefined => (
-            <div key={t.name}>
-              {t.classId !== null && (
-                <div className={styles.ClassItem}>
-                  <div className={styles.Color} style={{ backgroundColor: t.color }}>
+          {classes.map(({ name, classId, color }): ReactElement | undefined => (
+            <div key={name}>
+              {classId !== null && (
+                <button
+                  type="button"
+                  className={styles.ClassItem}
+                  onClick={() => newGroup(name.split(':')[0])}
+                >
+                  <div className={styles.Color} style={{ backgroundColor: color }}>
                     {' '}
                   </div>
-                  <p>{t.name.split(':')[0]}</p>
-                </div>
+                  <p>{name.split(':')[0]}</p>
```

Comment:
I'm not sure what is `name.split(':')[0]` doing. Maybe you can store it into a variable and give the variable a explanatory name?

Reply:
I think this is getting the course department and number like get `CS 3410` from `CS 3410: Systems Programming` but yes new variable name would be more readable 👍

Reply:
done